### DISCUSSION TITLE
Arduino Mega OTA Update along with serial comms

### DISCUSSION THREAD

**Discussion by aryadas98** [JAndrassy/ArduinoOTA#136] [General]
Hi, I want to send packets of data to and from the Mega over WiFi and preferably support OTA update with it. The OTA update works, and the examples from the WiFiEspAT library work. But from the documentation, it seems that AT only supports 1 TCP server, which is taken up by ArduinoOTA. So, any suggestions for how to establish communications?

**Comment by JAndrassy** [comment]
yes, that is [a very strange limitation](https://github.com/espressif/esp-at/issues/513#issuecomment-812905401) of the AT firmware (but the first versions of esp AT firmware had only one TCP connection!).
I too want to use the WiFiEspAT library in my project with multiple servers and OTA.
I wanted to solve this problem by adding server support to Jiri Bilek's AT firmware and make it multi-server. I did it, but the firmware doesn't perform well. It is unable to transport large files (yes this one too). I want to fix it, but I didn't get to it yet. 
Other option I was thinking to do, is to add support in ArduinoOTA to call the upload receiving function from sketch to allow its integration in  web server.
What you can do in your sketch is to integrate the InternalStorage into your web server similar to bin download in "Advanced" examples.

**Comment by aryadas98** [comment]
Hi, I wanted to say that I have solved this issue by using https://github.com/jandrassy/lab/blob/master/AvrDudeTelnet/AvrDudeTelnet.ino It's much simpler and easy to use, and even works on Windows 11 now. I modified it to have 2 ports - a comms port and a flashing port. Both do the same thing, except the flashing port resets the board before sending over the data.
